src/Lookup/SiteLinkLookup.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Lookup;
+
+use Wikibase\DataModel\Entity\ItemId;
+use Wikibase\DataModel\SiteLink;
+
+/**
+ * Contains methods to lookup of sitelinks of lookup by sitelinks.
+ *
+ * @since 2.0
+ *
+ * @licence GNU GPL v2+
+ * @author Jeroen De Dauw < jeroendedauw@gmail.com >
+ */
+interface SiteLinkLookup {
+
+	/**
+	 * Returns the id of the item that is equivalent to the
+	 * provided page, or null if there is none.
+	 *
+	 * @since 2.0
+	 *
+	 * @param string $globalSiteId
+	 * @param string $pageTitle
+	 *
+	 * @return ItemId|null
+	 */
+	public function getItemIdForLink( $globalSiteId, $pageTitle );
+
+	/**
+	 * Returns the links that match the provided conditions.
+	 * The links are returned as arrays with the following elements in specified order:
+	 * - string siteId
+	 * - string pageName
+	 * - int itemId Numeric (unprefixed) item id
```

Comment:
This is a weird interface. An array with three numerically indexed elements? I do not think such an interface should be in this generic services component. This sounds like a concept that's specific to Wikibase.git and should stay there.